### PR TITLE
Google sign in

### PR DESCRIPTION
**Pull request by stevegardiner26**
- Added Google Sign In Button in React
- Added a Logout Button to Log a User Out
- Handled creating users in the database or fetching existing users
- Handled User State between refreshes

(This PR was branched off of #3 so we need to review and merge that one before this one)

### CODE REVIEW

**Review by VivekSreenivasan — COMMENTED** [review]
Looks good

File: models/User.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const mongoose = require('mongoose');
+const {Schema} = mongoose;
+
+const userSchema = new Schema({
+    name: String,
+    email: String,
+    profile_image: String,
+    google_id: { type: String, unique: true }
```
Comment (VivekSreenivasan) [inline models/User.js:8]: I don't know if we can store the Google ID

**Review by VivekSreenivasan — COMMENTED** [review]
Oauth looks clean

**Review by jtheletter6 — APPROVED** [review]

**Review by VivekSreenivasan — APPROVED** [review]
Looks good

**Review by PRamos1196 — APPROVED** [review]
Looks Good

**Pull request merged**